sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -562,13 +577,27 @@ case class IntegralDivide(
       case d: DecimalType =>
         d.asIntegral.asInstanceOf[Integral[Any]]
     }
-    (x, y) => {
-      val res = integral.quot(x, y)
-      if (res == null) {
-        null
-      } else {
-        integral.asInstanceOf[Integral[Any]].toLong(res)
+    val _div =
+      (x: Any, y: Any) => {
+        val res = integral.quot(x, y)
+        if (res == null) {
+          null
+        } else {
+          integral.asInstanceOf[Integral[Any]].toLong(res)
+        }
       }
+
+    dataType match {
+      case LongType if failOnError =>
+        (x: Any, y: Any) => {
+          if (x == Long.MinValue && y == -1) {
```

Comment:
Shall we do this in `DivModLike.eval`?

Reply:
This is not happening in Divide or Mod

Reply:
yea, similar to the codegen path we can use `if (isIntegralDivide)`.

Reply:
Yeah it's tricky. It is hard to reuse the`doGenCode` here so I added a new flag `isIntegralDivide`. 
The original idea is to keep the special handling code in `IntegralDivide` if possible.
But now it seems asymmetric in both sides..